diff -urNp db-4.5.20.orig/crypto/aes_method.c db-4.5.20.fix/crypto/aes_method.c
--- db-4.5.20.orig/crypto/aes_method.c	2006-08-24 14:56:11.000000000 +0000
+++ db-4.5.20.fix/crypto/aes_method.c	2015-06-02 05:55:12.179693441 +0000
@@ -264,6 +264,6 @@ __aes_err(dbenv, err)
 		errstr = "AES error unrecognized";
 		break;
 	}
-	__db_errx(dbenv, errstr);
+	__db_errx(dbenv, "%s", errstr);
 	return;
 }
diff -urNp db-4.5.20.orig/repmgr/repmgr_net.c db-4.5.20.fix/repmgr/repmgr_net.c
--- db-4.5.20.orig/repmgr/repmgr_net.c	2006-09-20 00:20:58.000000000 +0000
+++ db-4.5.20.fix/repmgr/repmgr_net.c	2015-06-02 05:51:40.090693441 +0000
@@ -954,7 +954,7 @@ __repmgr_listen(dbenv)
 	}
 
 	ret = net_errno;
-	__db_err(dbenv, ret, why);
+	__db_err(dbenv, ret, "%s", why);
 clean:	if (s != INVALID_SOCKET)
 		(void)closesocket(s);
 	return (ret);
diff -urNp db-4.5.20.orig/rpc_client/client.c db-4.5.20.fix/rpc_client/client.c
--- db-4.5.20.orig/rpc_client/client.c	2006-08-24 14:56:16.000000000 +0000
+++ db-4.5.20.fix/rpc_client/client.c	2015-06-02 06:11:26.970928441 +0000
@@ -64,7 +64,7 @@ __dbcl_env_set_rpc_server(dbenv, clnt, h
 	if (clnt == NULL) {
 		if ((cl = clnt_create((char *)host, DB_RPC_SERVERPROG,
 		    DB_RPC_SERVERVERS, "tcp")) == NULL) {
-			__db_errx(dbenv, clnt_spcreateerror((char *)host));
+			__db_errx(dbenv, "%s", clnt_spcreateerror((char *)host));
 			return (DB_NOSERVER);
 		}
 		if (tsec != 0) {
diff -urNp db-4.5.20.orig/rpc_client/gen_client.c db-4.5.20.fix/rpc_client/gen_client.c
--- db-4.5.20.orig/rpc_client/gen_client.c	2006-09-20 20:07:21.000000000 +0000
+++ db-4.5.20.fix/rpc_client/gen_client.c	2015-06-02 06:13:36.256928441 +0000
@@ -81,7 +81,7 @@ __dbcl_env_create(dbenv, timeout)
 
 	replyp = __db_env_create_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -115,7 +115,7 @@ __dbcl_env_cdsgroup_begin(dbenv, txnpp)
 
 	replyp = __db_env_cdsgroup_begin_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -150,7 +150,7 @@ __dbcl_env_close(dbenv, flags)
 
 	replyp = __db_env_close_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -201,7 +201,7 @@ __dbcl_env_dbremove(dbenv, txnp, name, s
 
 	replyp = __db_env_dbremove_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -257,7 +257,7 @@ __dbcl_env_dbrename(dbenv, txnp, name, s
 
 	replyp = __db_env_dbrename_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -294,7 +294,7 @@ __dbcl_env_get_cachesize(dbenv, gbytesp,
 
 	replyp = __db_env_get_cachesize_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -334,7 +334,7 @@ __dbcl_env_get_encrypt_flags(dbenv, flag
 
 	replyp = __db_env_get_encrypt_flags_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -370,7 +370,7 @@ __dbcl_env_get_flags(dbenv, flagsp)
 
 	replyp = __db_env_get_flags_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -406,7 +406,7 @@ __dbcl_env_get_home(dbenv, homep)
 
 	replyp = __db_env_get_home_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -442,7 +442,7 @@ __dbcl_env_get_open_flags(dbenv, flagsp)
 
 	replyp = __db_env_get_open_flags_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -486,7 +486,7 @@ __dbcl_env_open(dbenv, home, flags, mode
 
 	replyp = __db_env_open_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -526,7 +526,7 @@ __dbcl_env_remove(dbenv, home, flags)
 
 	replyp = __db_env_remove_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -566,7 +566,7 @@ __dbcl_env_set_cachesize(dbenv, gbytes,
 
 	replyp = __db_env_set_cachesize_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -606,7 +606,7 @@ __dbcl_env_set_encrypt(dbenv, passwd, fl
 
 	replyp = __db_env_set_encrypt_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -643,7 +643,7 @@ __dbcl_env_set_flags(dbenv, flags, onoff
 
 	replyp = __db_env_set_flags_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -685,7 +685,7 @@ __dbcl_env_txn_begin(dbenv, parent, txnp
 
 	replyp = __db_env_txn_begin_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -725,7 +725,7 @@ __dbcl_env_txn_recover(dbenv, preplist,
 
 	replyp = __db_env_txn_recover_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -761,7 +761,7 @@ __dbcl_db_create(dbp, dbenv, flags)
 
 	replyp = __db_db_create_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -817,7 +817,7 @@ __dbcl_db_associate(dbp, txnp, sdbp, fun
 
 	replyp = __db_db_associate_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -857,7 +857,7 @@ __dbcl_db_close(dbp, flags)
 
 	replyp = __db_db_close_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -903,7 +903,7 @@ __dbcl_db_cursor(dbp, txnp, dbcpp, flags
 
 	replyp = __db_db_cursor_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -955,7 +955,7 @@ __dbcl_db_del(dbp, txnp, key, flags)
 
 	replyp = __db_db_del_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1014,7 +1014,7 @@ __dbcl_db_get(dbp, txnp, key, data, flag
 
 	replyp = __db_db_get_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1053,7 +1053,7 @@ __dbcl_db_get_bt_minkey(dbp, minkeyp)
 
 	replyp = __db_db_get_bt_minkey_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1096,7 +1096,7 @@ __dbcl_db_get_dbname(dbp, filenamep, dbn
 
 	replyp = __db_db_get_dbname_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1139,7 +1139,7 @@ __dbcl_db_get_encrypt_flags(dbp, flagsp)
 
 	replyp = __db_db_get_encrypt_flags_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1180,7 +1180,7 @@ __dbcl_db_get_flags(dbp, flagsp)
 
 	replyp = __db_db_get_flags_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1221,7 +1221,7 @@ __dbcl_db_get_h_ffactor(dbp, ffactorp)
 
 	replyp = __db_db_get_h_ffactor_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1262,7 +1262,7 @@ __dbcl_db_get_h_nelem(dbp, nelemp)
 
 	replyp = __db_db_get_h_nelem_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1303,7 +1303,7 @@ __dbcl_db_get_lorder(dbp, lorderp)
 
 	replyp = __db_db_get_lorder_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1344,7 +1344,7 @@ __dbcl_db_get_open_flags(dbp, flagsp)
 
 	replyp = __db_db_get_open_flags_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1385,7 +1385,7 @@ __dbcl_db_get_pagesize(dbp, pagesizep)
 
 	replyp = __db_db_get_pagesize_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1426,7 +1426,7 @@ __dbcl_db_get_q_extentsize(dbp, extentsi
 
 	replyp = __db_db_get_q_extentsize_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1467,7 +1467,7 @@ __dbcl_db_get_re_delim(dbp, delimp)
 
 	replyp = __db_db_get_re_delim_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1508,7 +1508,7 @@ __dbcl_db_get_re_len(dbp, lenp)
 
 	replyp = __db_db_get_re_len_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1549,7 +1549,7 @@ __dbcl_db_get_re_pad(dbp, padp)
 
 	replyp = __db_db_get_re_pad_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1605,7 +1605,7 @@ __dbcl_db_join(dbp, curs, dbcp, flags)
 	replyp = __db_db_join_4005(&msg, cl);
 	__os_free(dbenv, msg.curs.curs_val);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1659,7 +1659,7 @@ __dbcl_db_key_range(dbp, txnp, key, rang
 
 	replyp = __db_db_key_range_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1719,7 +1719,7 @@ __dbcl_db_open(dbp, txnp, name, subdb, t
 
 	replyp = __db_db_open_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1786,7 +1786,7 @@ __dbcl_db_pget(dbp, txnp, skey, pkey, da
 
 	replyp = __db_db_pget_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1845,7 +1845,7 @@ __dbcl_db_put(dbp, txnp, key, data, flag
 
 	replyp = __db_db_put_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1896,7 +1896,7 @@ __dbcl_db_remove(dbp, name, subdb, flags
 
 	replyp = __db_db_remove_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1952,7 +1952,7 @@ __dbcl_db_rename(dbp, name, subdb, newna
 
 	replyp = __db_db_rename_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -1992,7 +1992,7 @@ __dbcl_db_set_bt_minkey(dbp, minkey)
 
 	replyp = __db_db_set_bt_minkey_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2037,7 +2037,7 @@ __dbcl_db_set_encrypt(dbp, passwd, flags
 
 	replyp = __db_db_set_encrypt_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2077,7 +2077,7 @@ __dbcl_db_set_flags(dbp, flags)
 
 	replyp = __db_db_set_flags_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2117,7 +2117,7 @@ __dbcl_db_set_h_ffactor(dbp, ffactor)
 
 	replyp = __db_db_set_h_ffactor_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2157,7 +2157,7 @@ __dbcl_db_set_h_nelem(dbp, nelem)
 
 	replyp = __db_db_set_h_nelem_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2197,7 +2197,7 @@ __dbcl_db_set_lorder(dbp, lorder)
 
 	replyp = __db_db_set_lorder_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2237,7 +2237,7 @@ __dbcl_db_set_pagesize(dbp, pagesize)
 
 	replyp = __db_db_set_pagesize_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2277,7 +2277,7 @@ __dbcl_db_set_q_extentsize(dbp, extentsi
 
 	replyp = __db_db_set_q_extentsize_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2317,7 +2317,7 @@ __dbcl_db_set_re_delim(dbp, delim)
 
 	replyp = __db_db_set_re_delim_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2357,7 +2357,7 @@ __dbcl_db_set_re_len(dbp, len)
 
 	replyp = __db_db_set_re_len_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2397,7 +2397,7 @@ __dbcl_db_set_re_pad(dbp, pad)
 
 	replyp = __db_db_set_re_pad_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2443,7 +2443,7 @@ __dbcl_db_stat(dbp, txnp, sp, flags)
 
 	replyp = __db_db_stat_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2483,7 +2483,7 @@ __dbcl_db_sync(dbp, flags)
 
 	replyp = __db_db_sync_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2530,7 +2530,7 @@ __dbcl_db_truncate(dbp, txnp, countp, fl
 
 	replyp = __db_db_truncate_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2568,7 +2568,7 @@ __dbcl_dbc_c_close(dbc)
 
 	replyp = __db_dbc_c_close_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2609,7 +2609,7 @@ __dbcl_dbc_c_count(dbc, countp, flags)
 
 	replyp = __db_dbc_c_count_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2649,7 +2649,7 @@ __dbcl_dbc_c_del(dbc, flags)
 
 	replyp = __db_dbc_c_del_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2690,7 +2690,7 @@ __dbcl_dbc_c_dup(dbc, dbcp, flags)
 
 	replyp = __db_dbc_c_dup_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2744,7 +2744,7 @@ __dbcl_dbc_c_get(dbc, key, data, flags)
 
 	replyp = __db_dbc_c_get_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2805,7 +2805,7 @@ __dbcl_dbc_c_pget(dbc, skey, pkey, data,
 
 	replyp = __db_dbc_c_pget_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2859,7 +2859,7 @@ __dbcl_dbc_c_put(dbc, key, data, flags)
 
 	replyp = __db_dbc_c_put_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2897,7 +2897,7 @@ __dbcl_txn_abort(txnp)
 
 	replyp = __db_txn_abort_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2937,7 +2937,7 @@ __dbcl_txn_commit(txnp, flags)
 
 	replyp = __db_txn_commit_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -2977,7 +2977,7 @@ __dbcl_txn_discard(txnp, flags)
 
 	replyp = __db_txn_discard_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
@@ -3017,7 +3017,7 @@ __dbcl_txn_prepare(txnp, gid)
 
 	replyp = __db_txn_prepare_4005(&msg, cl);
 	if (replyp == NULL) {
-		__db_errx(dbenv, clnt_sperror(cl, "Berkeley DB"));
+		__db_errx(dbenv, "%s", clnt_sperror(cl, "Berkeley DB"));
 		ret = DB_NOSERVER;
 		goto out;
 	}
